validator/src/validator.rs

```diff
@@ -467,6 +478,9 @@ where
                 self.on_blockchain_extended(hash);
             }
             BlockchainEvent::EpochFinalized(ref hash) => {
+                // Reset the validator health for the new epoch
+                self.validator_health.write().reset_epoch();
```

Comment:
Why would everything get reset once an epoch finalises? The 'health' of a validator doesn't suddenly improve if a new epoch starts.